gno.land/pkg/gnoland/types.go

```diff
@@ -13,6 +18,44 @@ func ProtoGnoAccount() std.Account {
 }
 
 type GnoGenesisState struct {
-	Balances []string `json:"balances"`
-	Txs      []std.Tx `json:"txs"`
+	Balances []Balance `json:"balances"`
+	Txs      []std.Tx  `json:"txs"`
+}
+
+type Balance struct {
+	Address bft.Address
+	Value   std.Coins
+}
+
+func (b *Balance) Parse(line string) error {
+	parts := strings.Split(strings.TrimSpace(line), "=") // <address>=<coins>
+	if len(parts) != 2 {
+		return fmt.Errorf("invalid balance line: %q", line)
+	}
+
+	var err error
+
+	b.Address, err = crypto.AddressFromBech32(parts[0])
+	if err != nil {
+		return fmt.Errorf("invalid balance addr %s: %w", parts[0], err)
+	}
+
+	b.Value, err = std.ParseCoins(parts[1])
+	if err != nil {
+		return fmt.Errorf("invalid balance coins %s: %w", parts[1], err)
+	}
+
+	return nil
+}
+
+func (b *Balance) UnmarshalJSON(data []byte) error {
```

Comment:
is data JSON or `addr=bal`?

Reply:
I wanted to keep the previous format of the genesis state for compatibility reasons because the it was previously stored as `[]string`:
```diff
modified   gno.land/pkg/gnoland/types.go
@@ -16,25 +19,53 @@ func ProtoGnoAccount() std.Account {
 }
 
 type GnoGenesisState struct {
-	Balances []string `json:"balances"`
-	Txs      []std.Tx `json:"txs"`
+	Balances []Balance `json:"balances"`
+	Txs      []std.Tx  `json:"txs"`
```

Reply:
Remove or name them better.

Reply:
I made a mistake; I didn't realize it involved Amino Marshaling, so I updated the `MarshalJSON` method to align with Amino's `MarshalAmino` method and added some tests along the way. However, this does not change the fact that the application will break if it attempts to load a genesis state from before this PR without the `MarshalAmino` / `UnmarshalAmino` methods. 
Unless I revert `[]Balance` back to `[]string` or ignore this backward compatibility issue, but I don't really like those ideas.